<launch>
    <!-- args -->
    <arg name="start_rviz" default="true"/>

    <node pkg="path_planning" type="path_planning" name="path_planning" output="screen">
        <!-- plan env -->
        <param name="plan_env/odom_topic" value="/firefly/odometry_sensor1/odometry" type="string"/>
        <param name="pc2_topic" value="/livox/scan" type="string"/>

        <!-- grid map -->
        <param name="grid_map/local_update_range_x" value="5.5" type="double"/>
        <param name="grid_map/local_update_range_y" value="5.5" type="double"/>
        <param name="grid_map/local_update_range_z" value="1.5" type="double"/>
        <param name="grid_map/expansion_range" value="0.15" type="double"/>

        <!-- rrt -->
        <param name="rrt/max_iter" value="1000" type="int"/>

        <!-- path optimization -->
        <param name="path_optimization/limit_d" value="0.1" type="double"/>
        <param name="path_optimization/limit_vel" value="0.5" type="double"/>
        <param name="path_optimization/limit_acc" value="1.0" type="double"/>
        <param name="path_optimization/limit_jerk" value="1.0" type="double"/>
        <param name="path_optimization/traj_pose_topic" value="/traj_pose" type="string"/>

        <!-- publish the point of the trajectory -->
        <param name="path_optimization/traj_pose_topic" value="/firefly/command/pose" type="string"/>
        <param name="path_optimization/traj_multi_topic" value="/firefly/command/trajectory" type="string"/>
    </node>

    <group if="$(arg start_rviz)">
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find path_planning)/config/rviz.rviz" output="screen"/>
    </group>
</launch>